### DISCUSSION TITLE
Old version HAA update?

### DISCUSSION THREAD

**Discussion by Waniaok** [RavenSystem/esp-homekit-devices#1567] [General]
Hi, 
I have many haa device with old version (1.2.1, 1.3, 1,5.4, 2.5.1, 3.1.1, 3.5.11, 4.9)

how i can update tham?

OK
sent OK
HTTP returns 302
--- ota_connect LocalPort=d0c0 DNS IP:185.199.109.154 local..OK remote..OK SSL..OK set_fd to github-releases.githubusercontent.com port 443..failed, return [-0x1]
wolfSSL_send error = -322

--- ota_boot...1
--- running ota-main software
--- ota_compare 2.1.2 with 1.0.0=1
--- ota_get_hash
--- ota_get_file_ex
GET /HomeACcessoryKid/life-cycle-manager/releases/download/2.1.2/otaboot.bin.sig HTTP/1.1
Host: github.com

--- ota_connect LocalPort=d0c1 DNS IP:140.82.121.3 local..OK remote..OK SSL..OK set_fd to github.com port 443..OK
sent OK
HTTP returns 302
--- ota_connect LocalPort=d0c2 DNS IP:185.199.109.154 local..OK remote..OK SSL..OK set_fd to github-releases.githubusercontent.com port 443..failed, return [-0x1]
wolfSSL_send error = -322

--- ota_get_file
--- ota_get_file_ex
GET /HomeACcessoryKid/life-cycle-manager/releases/download/2.1.2/otaboot.bin HTTP/1.1
Host: github.com

--- ota_connect LocalPort=d0c3 DNS IP:140.82.121.3 local..OK remote..OK SSL..OK set_fd to github.com port 443..OK
sent OK
HTTP returns 302
--- ota_connect LocalPort=d0c4 DNS IP:185.199.109.154 local..OK remote..OK SSL..OK set_fd to github-releases.githubusercontent.com port 443..failed, return [-0x1]
wolfSSL_send error = -322

--- entering the loop
--- ota_get_pubkey
ret: 0
--- ota_set_verify...OFF
--- ota_get_version
--- ota_connect LocalPort=d0c5 DNS IP:140.82.121.3 local..OK remote..OK SSL..OK set_fd to github.com port 443..GET /HomeACcessoryKid/life-cycle-manager/releases/download/2.1.2/certs.sector.sig HTTP/1.1
Host: github.com

--- ota_connect LocalPort=d0c6 DNS IP:140.82.121.3 local..OK remote..OK SSL..OK set_fd to github.com port 443..OK
sent OK
HTTP returns 302
--- ota_connect LocalPort=d0c7 DNS IP:185.199.109.154 local..OK remote..OK SSL..OK set_fd to github-releases.githubusercontent.com port 443..OK
 collected 155 bytes
--- ota_verify_hash
--- ota_hash
--- ota_set_verify...ON
certs size: 2628
TIME: Sun Sep 19 07:10:08 2021
--- ota_get_version
--- ota_connect LocalPort=d0c8 DNS IP:140.82.121.3 local..OK remote..OK SSL..OK set_fd to github.com port 443..OK
GET /HomeACcessoryKid/life-cycle-manager/releases/latest HTTP/1.1
Host: github.com

HTTP returns 302 for --- ota_boot...1
HomeACcessoryKid/life-cycle-manager@version:"2.1.2"
--- ota_get_hash
--- ota_get_file_ex
GET /HomeACcessoryKid/life-cycle-manager/releases/download/2.1.2/certs.sector.sig HTTP/1.1
Host: github.com

--- ota_connect LocalPort=d0c9 DNS IP:140.82.121.3 local..OK remote..OK
sent OK
HTTP returns 302
--- ota_connect LocalPort=d0ca DNS IP:185.199.109.154 local..OK remote..OK SSL..OK set_fd to github-releases.githubusercontent.com port 443..failed, return [-0x1]
wolfSSL_send error = -322

--- ota_boot...1
--- running ota-main software
--- ota_compare 2.1.2 with 1.0.0=1
--- ota_get_hash
--- ota_get_file_ex
failed, return [-0x1]
wolfSSL_send error = -322

--- ota_boot...1
--- running ota-main software
--- ota_compare--- ota_connect LocalPort=d0cb DNS IP:140.82.121.3 local..OK remote..OK SSL..OK set_fd to github.com port 443..OK
sent OK
HTTP returns 302
--- ota_connect LocalPort=d0cc DNS IP:185.199.109.154 local..OK remote..OK SSL..OK set_fd to github-releases.githubusercontent.com port 443..failed, return [-0x1]
wolfSSL_send error = -322

--- ota_get_file
--- ota_get_file_ex
GET /HomeACcessoryKid/life-cycle-manager/releases/download/2.1.2/otaboot.bin HTTP/1.1
Host: github.com

--- ota_connect LocalPort=d0cd DNS IP:140.82.121.3 local..OK remote..OK SSL..OK set_fd to github.com port 443..OK
sent OK
HTTP returns 302
--- ota_connect LocalPort=d0ce DNS IP:185.199.109.154 local..OK remote..OK SSL..OK set_fd to github-releases.githubusercontent.com port 443..failed, return [-0x1]
wolfSSL_send error = -322

--- entering the loop

**Comment by buarle** [comment]
Hi,

I had similar issues in the past.
From my opinion:

for all devices within firmware (1.2.1, 1.3, 1,5.4, 2.5.1, 3.1.1, 3.5.11), you will need to re-flash them, since there was a breaking server-side change (something with casesensitivty) from GitHub.

thats why there is a possibility in firmware (4.9), where you should be able to set a local http server as download source. Just set up a tiny http server. (https://github.com/RavenSystem/esp-homekit-devices/wiki/Installation#using-a-custom-server-for-ota-updates)
I recently had issues with devices running 4.7.2 and the only way round was to update them localy to verison 6.0.0 and afterwards over the regular way towards 7.6.0
Jumping directly to 7.6.0 through the local server shoul work as well.

Best
Markus

**Comment by Waniaok** [comment]
thank you for your post.  I wanted to avoid disconnecting devices and flashing again, but since it's inevitable then it has to be done.
